tests/devices/qubit/test_apply_operation.py

```diff
@@ -800,7 +801,7 @@ def test_batch_size_set_if_missing(self, method, ml_framework):
         param = qml.math.asarray([0.1, 0.2, 0.3], like=ml_framework)
         state = np.ones((2, 2)) / 2
         op = qml.RX(param, 0)
-        op._batch_size = None  # pylint:disable=protected-access
+        assert op._batch_size == _UNSET_BATCH_SIZE  # pylint:disable=protected-access
```

Comment:
Throughout the tests, do we actually want to use `is` over `==` for testing against `_UNSET_BATCH_SIZE`? :) 
```suggestion
        assert op._batch_size is _UNSET_BATCH_SIZE  # pylint:disable=protected-access
```

Reply:
They would be the same for either `-1` or `object()`.

Reply:
Alright :) I was thinking about pylint asking us to use `is` over `==` when comparing to `True` or `False` in a test. Are they different for these Boolean values? I guess this depends on what one is comparing to those Booleans...

Reply:
Yeah, small integers are hashed. Larger ones are not:
```python
>>> x = 1000
>>> y = 1000
>>> id(x) == id(y)
False
```

Reply:
ok I don't like that... going to use `==/!=` instead of `is/is not`

Reply:
Sorry :upside_down_face:

Reply:
jk! i used `is` because `==` breaks tf.function tests, whereas `is` solves all problems magically! I'm putting it back lol. tests will fail if somehow this assumption is broken.